### PR TITLE
Add support for build tracing

### PR DESCRIPTION
**Pull request by sauyon**
c.f. #170

### CODE REVIEW

File: extractor/cli/go-extractor/go-extractor.go [hunk 1]
```diff
@@ -20,23 +20,70 @@ func usage() {
...
-		i++
+		// parse go build flags
+		switch args[i] {
+		// skip `-o output` and `-i`, if applicable
+		case "-o":
+			if i+1 < len(args) {
+				i++
+			}
+		case "-i":
+		case "-p", "-asmflags", "-buildmode", "-compiler", "-gccgoflags", "-gcflags", "-installsuffix",
```
Comment (sauyon) [inline extractor/cli/go-extractor/go-extractor.go:71]: > How did you come up with this list, and how do we maintain it going forward?

I used the documentation from `go help build`. Unfortunately I think we'll have to maintain it manually as there are new releases.

File: extractor/cli/go-extractor/go-extractor.go [hunk 2]
```diff
@@ -20,23 +20,70 @@ func usage() {
...
+				os.Exit(0)
+			case "--mimic":
+				if i+1 < len(args) {
+					i++
+					compiler := args[i]
+					log.Printf("Compiler: %s", compiler)
+					if i+1 < len(args) {
+						i++
+						command := args[i]
+						if command == "build" || command == "install" {
```
Comment (sauyon) [inline extractor/cli/go-extractor/go-extractor.go:46]: > What about `"run"`? I think that also triggers compilation.

Right.

File: extractor/cli/go-extractor/go-extractor.go [hunk 3]
```diff
@@ -20,23 +20,70 @@ func usage() {
...
 
-		i++
+		// parse go build flags
+		switch args[i] {
+		// skip `-o output` and `-i`, if applicable
+		case "-o":
+			if i+1 < len(args) {
+				i++
+			}
+		case "-i":
```
Comment (sauyon) [inline extractor/cli/go-extractor/go-extractor.go:70]: > It looks like `-i` isn't actually skipped?

It is, as it's not added to the build flags list. I could put a continue here to make it more clear, though?

File: extractor/cli/go-extractor/go-extractor.go [hunk 4]
```diff
@@ -20,23 +20,70 @@ func usage() {
...
+		case "-p", "-asmflags", "-buildmode", "-compiler", "-gccgoflags", "-gcflags", "-installsuffix",
+			"-ldflags", "-mod", "-modfile", "-pkgdir", "-tags", "-toolexec":
+			if i+1 < len(args) {
+				buildFlags = append(buildFlags, args[i], args[i+1])
+				i++
+			} else {
+				buildFlags = append(buildFlags, args[i])
+			}
+		default:
+			if args[i][0] == '-' {
```
Comment (sauyon) [inline extractor/cli/go-extractor/go-extractor.go:80]: > Maybe make sure that `args[i]` isn't the empty string. Not sure if that can happen, but just to be safe.

Right. I'll just use `strings.HasPrefix`.

**Comment by sauyon** [comment]
[The fix](https://git.semmle.com/Semmle/code/pull/37108) for the MacOS issues has been merged; should I started testing on OS X again?

**Comment by max-schaefer** [comment]
Yes, please. I've not yet been able to find time to look at this again, I'm afraid.

**Comment by max-schaefer** [comment]
I tae it this is superseded by #324?

**Comment by sauyon** [comment]
Ah, right.

**Pull request closed without merging**